### ISSUE TITLE
Donate Option

### ISSUE THREAD

**Issue by mthxx** [mthxx/Budget#32]
Add a donation menu option

Clicking it should bring up a window that suggests links to various services such as

*Paypal
*Flattr
*Bitcoin

**Comment by thorsummoner** [comment]
You lost me with this one

**Comment by mthxx** [comment]
I'd like to provide people with an easy way to contribute financially if they want to.

If there in a menu option to donate, then they can just click on "Paypal" and it would load paypal in their browser and auto fill the recipent information with the appropriate paypal account. It's a bit more streamlined than having someone visit the projects website and try to find the donate link options there.

**Comment by thorsummoner** [comment]
I'm not opposing you and I know what it is. Merely in-all donation buttons lose my interest just as fast as in app advertisements. (Yet somehow I put up with Steam)

**Comment by mthxx** [comment]
Ahh, I see what you are saying. Do you have a suggested, less intrusive, implementation idea?

From my experience, a lot of FOSS projects don't include a "donate" option anywhere in their app. This forces people who actively donate (like myself) to search their project/organization websites. Some websites make it easier to find donate links than others. Is that extra step of searching for a projects website and then donation button preferable over having a donate option in the global menu?

**Comment by thorsummoner** [comment]
Do feel free to write in your donate button, though I feel you might have better (or additional) monetization come by adding a price point on the Ubuntu store, where people are already comfortable with spending money. 
Of you do include a donate button, some of the last annoying places to put it are in the help->about menu item, near the credits and other information that is not relevant to using the application. One thing to be weary of, you don't want to be annoying to users who may have become frustrated, searching for a way to do the thing they need to and have a handful of donation items "in their way". Recessing them where users look when they want that kind of irrelevant to app " because I use it all the time and it's worth checking for an easter egg " *feels* the most responsible to me personally as a FOSS enthusiast. 

Also you should have seen the spike of users who left Heidi SQL when their automatic updates added a big green and red donate button that was always visible on their menu bar. Absolutely horrendous

**Comment by mthxx** [comment]
Yes, I completely agree with you. Having a donate button that is obtrusive is defintely a bad idea. My idea was having it basically be the last line item in the global menu. So on Gnome, if you click the name of the application in the top left of the screen, the last item might be "Donate".

I like the idea of having it in the credits page. That would put it even more out of the way, people that are explicitly looking to donate might click that. Of course, the common practice would probably still be to visit the projects website.

**Comment by thorsummoner** [comment]
Yes, I have no dismay with moderately strong project website donation accessibility. Links to a project website are occasionally, if not often, present in the credits page as well.

Also, if a credits page is off the roadmap, I would be okay with a temporary donation button where you say. I'm really not used to this new wide header style of desktop app. Nore can I run your project as is due to three shipped version of gnome not having popovers.

-cheers and thanks for being open to discussion!

**Comment by mthxx** [comment]
What version of Gnome are you running? I thought popovers had been around a while. I can look into adding a non-popover method of doing things. Right now, there's only 2 different actions that bring up a popover. It might still be feasable to have a non-popover option for older versions of Gnome.

**Comment by thorsummoner** [comment]
I can't be sure which package version it is but my python(2 & 3) installed reports gi.repository._version of 3.0

**Comment by mthxx** [comment]
Can you open up a terminal and type:

gnome-shell --version


--
Marc Thomas
mthx.org | Github: mthxx | @mthx_

On Mon, Feb 2, 2015 at 10:17 AM, thorsummoner 
<notifications@github.com> wrote:
> I can't be sure which package version it is but my python(2 & 3) 
> installed reports gi.repository._version of 3.0
> 
> —
> Reply to this email directly or view it on GitHub.
> 
=

**Comment by thorsummoner** [comment]
Ah, sorry, I run cinnamon, no gnome shell on this system. I can't stand vanilla. Which ships with libgtk-3-0 3.10.8~8+quinta

**Comment by thorsummoner** [comment]
TBH, this conversation should be in a different thread; But I Have commented out all lines that are not compatible with the current versions of GTK shipped with Ubuntu 14/Cinnamon (Their almost certainly the exact same repositories), that is here: 

- Branch [legacy-release-compat](https://github.com/thorsummoner/Budget/tree/legacy-release-compat) 
- Permalink [a88bc1911a6f](https://github.com/thorsummoner/Budget/tree/a88bc1911a6fb75c9552a3075d856e2a8801e9b1)
